packages/amazon/src/serverGroup/details/scalingPolicy/upsert/ScalingPolicyCommandBuilderService.ts

```diff
@@ -0,0 +1,127 @@
+import { cloneDeep } from 'lodash';
+
+import {
+  ISimplePolicyDescription,
+  IStepPolicyDescription,
+  IUpsertAlarmDescription,
+  IUpsertScalingPolicyCommand,
+} from '../ScalingPolicyWriter';
+import { IAmazonServerGroup, IScalingPolicy, IStepAdjustment, ITargetTrackingPolicy } from '../../../../domain';
+
+type PolicyType = 'Step' | 'TargetTracking';
+
+export const ScalingPolicyCommandBuilder = {
+  buildAlarm: (policy: IScalingPolicy, region: string, asgName: string): IUpsertAlarmDescription => {
+    const alarm = policy?.alarms[0];
+    return {
+      name: alarm.alarmName,
+      actionsEnabled: true,
+      alarmActionArns: alarm.alarmActions,
+      alarmDescription: alarm.alarmDescription,
+      asgName: asgName,
+      comparisonOperator: alarm.comparisonOperator,
+      dimensions: alarm.dimensions,
+      evaluationPeriods: alarm.evaluationPeriods,
+      insufficientDataActionArns: alarm.insufficientDataActions,
+      metricName: alarm.metricName,
+      namespace: alarm.namespace,
+      okActionArns: alarm.okactions,
+      period: alarm.period,
+      region,
+      statistic: alarm.statistic,
+      threshold: alarm.threshold,
+      unit: alarm.unit,
+    };
+  },
+
+  buildStepPolicy: (policy: IScalingPolicy, threshold: number, cooldown: number): IStepPolicyDescription => {
+    const stepAdjustments = policy.stepAdjustments.map((adjustment) => {
+      const step = {
+        scalingAdjustment: Math.abs(adjustment.scalingAdjustment),
+      } as IStepAdjustment;
+
+      if (adjustment.metricIntervalUpperBound !== undefined) {
+        step.metricIntervalUpperBound = adjustment.metricIntervalUpperBound + threshold;
+      }
+      if (adjustment.metricIntervalLowerBound !== undefined) {
+        step.metricIntervalLowerBound = adjustment.metricIntervalLowerBound + threshold;
+      }
```

Comment:
I don't know how to interpret these numbers, but just wanted to check if this was supposed to be `metricIntervalLowerBound - threshold`

Reply:
No, according to the existing controller it is added: https://github.com/spinnaker/deck/blob/master/packages/amazon/src/serverGroup/details/scalingPolicy/upsert/upsertScalingPolicy.controller.js#L108

There is some subtraction that happens when the command is prepared for submit.